src/lib/crypto_params/gen/dune

```diff
@@ -31,7 +31,7 @@
    mina_metrics.none
    logger.fake
  )
- (forbidden_libraries node_config)
+ (forbidden_libraries mina_node_config)
```

Comment:
Why was this renamed?

Reply:
It wasn't renamed, just public name will be used instead of private name. Some context: Ocaml packages always have a private name and sometimes have a public name. When two packages are compiled simultaneously from their codes, both private and public names are good to go. This is the case of running `dune build src/lib` that we currently have in CI.

Nix wrapping introduced in this PR has a different strategy: it compiles every package in isolation with code of other packages not being available. Dependencies are supplied as precompiled units via `OCAMLPATH` variable. When package is used as a precompiled dependency from path, it only can be referred to by its public name. Hence the need to update this `dune` file.